synapse/handlers/federation.py

```diff
@@ -1305,6 +1308,7 @@ def get_next(it, opt=None):
             if current_local.event_id < current_remote.event_id:
                 missing_locals.append(current_local)
                 current_local = get_next(local_iter)
+                continue
```

Comment:
I believe this is a real bug, without this the below code can run, but with a new `current_local` (which could be `None`).

Reply:
unchecked since I don't know the code well

Reply:
`construct_auth_difference` doesn't seem to be used by anything anymore, so let's just delete this entire function?

Reply:
Good catch!